packages/next/src/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -465,7 +465,7 @@ function getCodeAnalyzer(params: {
           buildInfo.importLocByPath = new Map()
         }
 
-        const importedModule = node.source.value?.toString()!
+        const importedModule = node.source.value?.toString()
```

Comment:
https://typescript-eslint.io/rules/no-non-null-asserted-optional-chain: putting a `!` _("this must exist")_ after a `?.` _("this might not exist")_ is almost certainly a sign of something being wrong in the type system understanding of the code. Here, the `!` didn't do anything because `node` is type `any`.